// oceanic.js/dist/lib/routes/Miscellaneous.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
/** @module Routes/Miscellaneous */
const Routes = tslib_1.__importStar(require("../util/Routes"));
const Application_1 = tslib_1.__importDefault(require("../structures/Application"));
const ClientApplication_1 = tslib_1.__importDefault(require("../structures/ClientApplication"));
/** Methods that don't fit anywhere else. */
class Miscellaneous {
    #manager;
    constructor(manager) {
        this.#manager = manager;
    }
    /**
     * Get the currently authenticated bot's application info.
     */
    async getApplication() {
        return this.#manager.authRequest({
            method: "GET",
            path: Routes.APPLICATION
        }).then(data => new Application_1.default(data, this.#manager.client));
    }
    /**
     * Get the currently authenticated bot's application info as a bare {@link ClientApplication~ClientApplication | ClientApplication}.
     */
    async getClientApplication() {
        return this.#manager.authRequest({
            method: "GET",
            path: Routes.APPLICATION
        }).then(data => new ClientApplication_1.default(data, this.#manager.client));
    }
    /**
     * Get the nitro sticker packs.
     */
    async getNitroStickerPacks() {
        return this.#manager.authRequest({
            method: "GET",
            path: Routes.NITRO_STICKER_PACKS
        }).then(data => data.sticker_packs.map(pack => ({
            bannerAssetID: pack.banner_asset_id,
            coverStickerID: pack.cover_sticker_id,
            description: pack.description,
            id: pack.id,
            name: pack.name,
            skuID: pack.sku_id,
            stickers: pack.stickers.map(sticker => this.#manager.client.util.convertSticker(sticker))
        })));
    }
    /**
     * Get a sticker.
     * @param stickerID The ID of the sticker to get.
     */
    async getSticker(stickerID) {
        return this.#manager.authRequest({
            method: "GET",
            path: Routes.STICKER(stickerID)
        }).then(data => this.#manager.client.util.convertSticker(data));
    }
    /**
     * Get the list of usable voice regions.
     */
    async getVoiceRegions() {
        return this.#manager.authRequest({
            method: "GET",
            path: Routes.VOICE_REGIONS
        });
    }
}
exports.default = Miscellaneous;
